source/errors.go

```diff
@@ -0,0 +1,12 @@
+package source
+
+// ErrDuplicateMigration is an error type for reporting duplicate migration
+// files.
+type ErrDuplicateMigration struct {
+	Filename string
```

Comment:
This should probably be a `source.Migration` and an `os.File` instead of a filename string. I'll fix this myself to avoid an extra review roundtrip